docs/custom_quantum_functions.md

```diff
@@ -202,14 +202,77 @@ This function gets called at the end of all QMK processing, before starting the
 
 Similar to `matrix_scan_*`, these are called as often as the MCU can handle. To keep your board responsive, it's suggested to do as little as possible during these function calls, potentially throtting their behaviour if you do indeed require implementing something special.
 
+### Example `void housekeeping_task_user(void)` implementation
+
+This example will show you how to use `void housekeeping_task_user(void)` to turn off RGB light. For RGB Matrix just use `#define RGB_MATRIX_TIMEOUT` in your `config.h`.
+
+First add those lines to your user `config.h`.
+
+```c
+#define RGBLIGHT_SLEEP  // enable rgblight_suspend() and rgblight_wakeup() in keymap.c
+#define RGBLIGHT_TIMEOUT 900000  // ms to wait until rgblight time out, 900K ms is 15min.
+```
+
+And add those lines to your `keymap.c`.
+
+```c
+static uint32_t key_timer;           // timer for last keyboard activity, use 32bit value and function to make longer idle time possible
+static void refresh_rgb(void);       // refreshes the activity timer and RGB, invoke whenever any activity happens
+static void check_rgb_timeout(void); // checks if enough time has passed for RGB to timeout
+bool is_rgb_timeout = false;         // store if RGB has timed out or not in a boolean
+
+void refresh_rgb()
+{
```

Comment:
Examples should follow coding style guidelines. 

Eg: 
```suggestion
void refresh_rgb(void) {
```

Reply:
Just made those changes, let me know if anything else I need change.